### DISCUSSION TITLE
Difficulty converting a BBT-managed RMarkdown document to docx and maintaining live Zotero citations

### DISCUSSION THREAD

**Discussion by TaymAlsalti** [retorquere/zotero-better-bibtex#2735] [Q&A]
Hi everyone,

I'm encountering problems trying to convert my .rmd file to a docx. with live Zotero citations:
When I run this in the Terminal within Rstudio and RProject (working dir)
`pandoc -s --lua-filter=zotero.lua --metadata=zotero_scannable_cite:true --metadata=zotero_client:jurism -o Thesis.docx -f markdown -t docx Thesis.rmd`
I get the error below, which seems to contain all my citation keys. I had my Zotero (v. 6.0.30) open. Other potentially relevant information: 

- pandoc v. 3.1.11
- zotero.lua (latest version) is in the working dir
- references.bib is in the working dir, although I get the same error with or without that line in the YAML header
- RStudio 2023.09.1 Build 494
- I originally knit the document to a PDF using the papaja package and managed all references using Zotero and Better Bibtex
- Abridged YAML header in Thesis.rmd:

```
---

bibliography        : ["references.bib"]
zotero:
  library: "tDCS-Motor Learning" # zotero group name
  client: zotero 
  author-in-text: false 
  csl-style: apa 
---
```

Would appreciate any hints!

Many thanks,
Taym

> zotero-live-citations caf41cc
Error running filter zotero.lua:
Could not fetch http://127.0.0.1:24119/better-bibtex/export/item?pandocFilterData=true&translator=json&citationKeys=dubljevicRisingTideTDCS2014%2CbanksPublicationBiasCall2012%2CevansHierarchyEvidenceFramework2003%2CstanleyMetaRegressionMethods2008%2CkepesMetaanalyticReviewsOrganizational2013%2CpagePRISMA2020Statement2021%2CmaassenReproducibilityIndividualEffect2020%2CR%2Dforcats%2CkangTranscranialDirectCurrent2016a%2CwurzmanOpenLetterConcerning2016%2CjonesHighPrevalenceLow2005%2CR%2Dmeta%2CzettlerTDCSResearchWorld2017%2CschalkenReportingQualitySystematic2017%2CbeggOperatingCharacteristicsRank1994%2CstanleyMetaregressionApproximationsReduce2014%2CshamseerPreferredReportingItems2015%2CrosenthalFileDrawerProblem1979%2CwexlerWhoUsesDirecttoConsumer2018%2CveveaPublicationBias2019%2CsimonsohnBetterPcurvesMaking2015%2CsimmonsFalsePositivePsychologyUndisclosed2011%2CaytugRevealedConcealedTransparency2012%2CcooperHandbookResearchSynthesis2009%2CxieDynamicDocumentsKnitr2015%2CgelmanGardenForkingPaths2013%2CgeyskensReviewEvaluationMetaAnalysis2009%2CkangTranscranialDirectCurrent2018%2CwanEstimatingSampleMean2014%2CpussegodaSystematicReviewAdherence2017%2CmoreauConductingMetaanalysisAge2020%2CviechtbauerOutlierInfluenceDiagnostics2010%2CReportingGuidelinesEQUATOR%2CluedtkeTranscranialDirectCurrent2012%2CborensteinCriticismsMetaanalysis2009%2Csimonsohn59PETPEESENot2017%2CR%2Dbase%2CR%2Dmetafor%2CadaImpactMetaanalyticDecisions2012a%2CnitscheTranscranialDirectCurrent2008%2CrstudioteamRStudioIntegratedDevelopment2021%2CsimonsohnPCurveEffectSize2014%2CmorenoAssessmentRegressionbasedMethods2009a%2CwoodsTechnicalGuideTDCS2016%2CtaylorTriangulatingMetaanalysesExample2016%2CR%2DMatrix%2CbuttonPowerFailureWhy2013%2CnosekPreregistrationRevolution2018%2CmoherImprovingQualityReports2000%2CliuEffectsTranscranialElectrical2021%2CcarterCorrectingBiasPsychology2019%2CR%2DMAd%2CtobiasAssessingInfluenceSingle1999a%2CfregniRegulatoryConsiderationsClinical2015%2CaustinPrefrontalElectricalStimulation2016%2CdavisRegulationConsumerTDCS2016%2CliberatiPRISMAStatementReporting2009%2CkonstantopoulosStatisticallyAnalyzingEffect2019%2CmcshaneAdjustingPublicationBias2016%2CR%2Dreadxl%2CR%2Dtidyr%2CgazzanigaMethodsCognitiveNeuroscience2018%2CveveaGeneralLinearModel1995%2CminarikImportanceSampleSize2016%2CmoherPreferredReportingItems2009%2CR%2Dreadr%2CdieckmannEmpiricalAssessmentMetaAnalytic2009%2CplesserReproducibilityVsReplicability2018%2CmadeyskiEffectSizesTheir2018%2CusheyRenvProjectEnvironments2021a%2CxieKnitrGeneralpurposePackage2021%2CR%2Dggplot2%2ChardwickeShouldPsychologyJournals2019%2ClakensExaminingReproducibilityMetaAnalyses2017%2CmorgantiImpactMetaanalysesClinical2007%2CriggsAtHomeTranscranialDirect2018%2CrohatgiWebPlotDigitizer2021%2CvanassenMetaanalysisUsingEffect2015%2CbuchEffectsTDCSMotor2017%2CgopalakrishnanSystematicReviewsMetaanalysis2013%2CeggerBiasMetaanalysisDetected1997%2CbakkerRulesGameCalled2012%2CrosenAnodalTDCSRight2016%2CR%2DkableExtra%2ClipkaResolvingHeterogeneityTranscranial2021%2CR%2Dstringr%2CnieminenMetaanalyticDecisionsReliability2011%2CwayantEvaluationReproducibleResearch2019%2ChigginsonCurrentIncentivesScientists2016%2CbrunoniTranscranialDirectCurrent2016%2CquintanaPreregistrationPublicationNontechnical2015%2CvoracekWhichDataMetaanalyze2019%2CbuscemiSingleDataExtraction2006%2ChyattQuandaryCovaryingBrief2020a%2ChungEfficacyTranscranialDirect2021%2ChozoEstimatingMeanVariance2005%2ChilgardOverstatedEvidenceShortterm2017%2CveveaPublicationBiasResearch2005%2CxieKnitrComprehensiveTool2014%2CpageEvaluationsUptakeImpact2017%2CR%2Dtibble%2CvanaertConductingMetaAnalysesBased2016%2ChorvathQuantitativeReviewFinds2015%2CpolaninTransparencyReproducibilityMetaAnalyses2020%2CpagePRISMA2020Explanation2021%2CgleserStochasticallyDependentEffect2009%2CmorrisCombiningEffectSize2002%2CsterlingPublicationDecisionsTheir1959%2CopensciencecollaborationEstimatingReproducibilityPsychological2015a%2CreisNoninvasiveCorticalStimulation2009%2CmoherPreferredReportingItems2015%2CR%2Dpapaja%2CguzzoMetaanalysisAnalysis1987%2CveronikiMethodsEstimateBetweenstudy2016%2CgebodhTranscranialDirectCurrent2019%2CduvalNonparametricTrimFill2000%2CfordErrorsConductSystematic2010%2CioannidisMassProductionRedundant2016%2CR%2Ddmetar%2ChungEfficacyTranscranialDirect2021a%2CR%2Ddplyr%2CgotzscheDataExtractionErrors2007%2CsteenbergenUnfocusFocUs2016%2CahnReviewMetaAnalysesEducation2012%2CgianniTDCSRandomizedControlled2021%2CstaggPhysiologicalBasisTranscranial2011%2CvalentineHowManyStudies2010%2CaguinisMetaAnalyticChoicesJudgment2011%2Cbotvinik%2DnezerVariabilityAnalysisSingle2020%2ChigginsCochraneHandbookSystematic2019%2CnosekPromotingOpenResearch2015%2CpageReproducibleResearchPractices2018%2ChohnEmpiricalReviewResearch2020%2CrothsteinPublicationBiasMetaanalysis2005%2CharrerDoingMetaAnalysisHandsOn2021%2CbromanRecommendationsFundingAgencies2017%2CsimonsohnPcurveKeyFiledrawer2014%2CborensteinEffectSizesMetaanalyses2019%2CborensteinPublicationBias2009%2CR%2Dpurrr%2CsteinerCausalReplicationFramework2019%2CgoodmanWhatDoesResearch2016%2CinzlichtBiasCorrectionTechniquesAlone2015%2CbennabiTranscranialDirectCurrent2014%2CaguinisDebunkingMythsUrban2011%2CheireneCallReplicationsAddiction2021
HttpExceptionRequest Request {
  host                 = "127.0.0.1"
  port                 = 24119
  secure               = False
  requestHeaders       = []
  path                 = "/better-bibtex/export/item"
  queryString          = "?pandocFilterData=true&translator=json&citationKeys=dubljevicRisingTideTDCS2014%2CbanksPublicationBiasCall2012%2CevansHierarchyEvidenceFramework2003%2CstanleyMetaRegressionMethods2008%2CkepesMetaanalyticReviewsOrganizational2013%2CpagePRISMA2020Statement2021%2CmaassenReproducibilityIndividualEffect2020%2CR%2Dforcats%2CkangTranscranialDirectCurrent2016a%2CwurzmanOpenLetterConcerning2016%2CjonesHighPrevalenceLow2005%2CR%2Dmeta%2CzettlerTDCSResearchWorld2017%2CschalkenReportingQualitySystematic2017%2CbeggOperatingCharacteristicsRank1994%2CstanleyMetaregressionApproximationsReduce2014%2CshamseerPreferredReportingItems2015%2CrosenthalFileDrawerProblem1979%2CwexlerWhoUsesDirecttoConsumer2018%2CveveaPublicationBias2019%2CsimonsohnBetterPcurvesMaking2015%2CsimmonsFalsePositivePsychologyUndisclosed2011%2CaytugRevealedConcealedTransparency2012%2CcooperHandbookResearchSynthesis2009%2CxieDynamicDocumentsKnitr2015%2CgelmanGardenForkingPaths2013%2CgeyskensReviewEvaluationMetaAnalysis2009%2CkangTranscranialDirectCurrent2018%2CwanEstimatingSampleMean2014%2CpussegodaSystematicReviewAdherence2017%2CmoreauConductingMetaanalysisAge2020%2CviechtbauerOutlierInfluenceDiagnostics2010%2CReportingGuidelinesEQUATOR%2CluedtkeTranscranialDirectCurrent2012%2CborensteinCriticismsMetaanalysis2009%2Csimonsohn59PETPEESENot2017%2CR%2Dbase%2CR%2Dmetafor%2CadaImpactMetaanalyticDecisions2012a%2CnitscheTranscranialDirectCurrent2008%2CrstudioteamRStudioIntegratedDevelopment2021%2CsimonsohnPCurveEffectSize2014%2CmorenoAssessmentRegressionbasedMethods2009a%2CwoodsTechnicalGuideTDCS2016%2CtaylorTriangulatingMetaanalysesExample2016%2CR%2DMatrix%2CbuttonPowerFailureWhy2013%2CnosekPreregistrationRevolution2018%2CmoherImprovingQualityReports2000%2CliuEffectsTranscranialElectrical2021%2CcarterCorrectingBiasPsychology2019%2CR%2DMAd%2CtobiasAssessingInfluenceSingle1999a%2CfregniRegulatoryConsiderationsClinical2015%2CaustinPrefrontalElectricalStimulation2016%2CdavisRegulationConsumerTDCS2016%2CliberatiPRISMAStatementReporting2009%2CkonstantopoulosStatisticallyAnalyzingEffect2019%2CmcshaneAdjustingPublicationBias2016%2CR%2Dreadxl%2CR%2Dtidyr%2CgazzanigaMethodsCognitiveNeuroscience2018%2CveveaGeneralLinearModel1995%2CminarikImportanceSampleSize2016%2CmoherPreferredReportingItems2009%2CR%2Dreadr%2CdieckmannEmpiricalAssessmentMetaAnalytic2009%2CplesserReproducibilityVsReplicability2018%2CmadeyskiEffectSizesTheir2018%2CusheyRenvProjectEnvironments2021a%2CxieKnitrGeneralpurposePackage2021%2CR%2Dggplot2%2ChardwickeShouldPsychologyJournals2019%2ClakensExaminingReproducibilityMetaAnalyses2017%2CmorgantiImpactMetaanalysesClinical2007%2CriggsAtHomeTranscranialDirect2018%2CrohatgiWebPlotDigitizer2021%2CvanassenMetaanalysisUsingEffect2015%2CbuchEffectsTDCSMotor2017%2CgopalakrishnanSystematicReviewsMetaanalysis2013%2CeggerBiasMetaanalysisDetected1997%2CbakkerRulesGameCalled2012%2CrosenAnodalTDCSRight2016%2CR%2DkableExtra%2ClipkaResolvingHeterogeneityTranscranial2021%2CR%2Dstringr%2CnieminenMetaanalyticDecisionsReliability2011%2CwayantEvaluationReproducibleResearch2019%2ChigginsonCurrentIncentivesScientists2016%2CbrunoniTranscranialDirectCurrent2016%2CquintanaPreregistrationPublicationNontechnical2015%2CvoracekWhichDataMetaanalyze2019%2CbuscemiSingleDataExtraction2006%2ChyattQuandaryCovaryingBrief2020a%2ChungEfficacyTranscranialDirect2021%2ChozoEstimatingMeanVariance2005%2ChilgardOverstatedEvidenceShortterm2017%2CveveaPublicationBiasResearch2005%2CxieKnitrComprehensiveTool2014%2CpageEvaluationsUptakeImpact2017%2CR%2Dtibble%2CvanaertConductingMetaAnalysesBased2016%2ChorvathQuantitativeReviewFinds2015%2CpolaninTransparencyReproducibilityMetaAnalyses2020%2CpagePRISMA2020Explanation2021%2CgleserStochasticallyDependentEffect2009%2CmorrisCombiningEffectSize2002%2CsterlingPublicationDecisionsTheir1959%2CopensciencecollaborationEstimatingReproducibilityPsychological2015a%2CreisNoninvasiveCorticalStimulation2009%2CmoherPreferredReportingItems2015%2CR%2Dpapaja%2CguzzoMetaanalysisAnalysis1987%2CveronikiMethodsEstimateBetweenstudy2016%2CgebodhTranscranialDirectCurrent2019%2CduvalNonparametricTrimFill2000%2CfordErrorsConductSystematic2010%2CioannidisMassProductionRedundant2016%2CR%2Ddmetar%2ChungEfficacyTranscranialDirect2021a%2CR%2Ddplyr%2CgotzscheDataExtractionErrors2007%2CsteenbergenUnfocusFocUs2016%2CahnReviewMetaAnalysesEducation2012%2CgianniTDCSRandomizedControlled2021%2CstaggPhysiologicalBasisTranscranial2011%2CvalentineHowManyStudies2010%2CaguinisMetaAnalyticChoicesJudgment2011%2Cbotvinik%2DnezerVariabilityAnalysisSingle2020%2ChigginsCochraneHandbookSystematic2019%2CnosekPromotingOpenResearch2015%2CpageReproducibleResearchPractices2018%2ChohnEmpiricalReviewResearch2020%2CrothsteinPublicationBiasMetaanalysis2005%2CharrerDoingMetaAnalysisHandsOn2021%2CbromanRecommendationsFundingAgencies2017%2CsimonsohnPcurveKeyFiledrawer2014%2CborensteinEffectSizesMetaanalyses2019%2CborensteinPublicationBias2009%2CR%2Dpurrr%2CsteinerCausalReplicationFramework2019%2CgoodmanWhatDoesResearch2016%2CinzlichtBiasCorrectionTechniquesAlone2015%2CbennabiTranscranialDirectCurrent2014%2CaguinisDebunkingMythsUrban2011%2CheireneCallReplicationsAddiction2021"
  method               = "GET"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
  proxySecureMode      = ProxySecureWithConnect
}
 (ConnectionFailure Network.Socket.connect: <socket: 768>: failed (Connection refused (WSAECONNREFUSED)))
stack traceback:
        zotero.lua:1680: in upvalue 'load_items'
        zotero.lua:1690: in function 'zotero.get'
        zotero.lua:1847: in function <zotero.lua:1830>
        (...tail calls...)

**Comment by retorquere** [comment]
If you still experience this, can you open an issue for it?

**Comment by TaymAlsalti** [comment]
[Is my confidence interval face so quickly forgettable? :(](https://github.com/retorquere/zotero-better-bibtex/issues/2736#issue-2060270980)

**Reply by retorquere** [reply]
I get a lot of messages, can't remember to correlate them all
